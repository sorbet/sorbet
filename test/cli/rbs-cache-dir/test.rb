# typed: true

class A
  #: () -> Integer
  def foo = 0
end
