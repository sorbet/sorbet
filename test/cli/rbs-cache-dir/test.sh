#!/bin/bash
set -euo pipefail

trap 'rm -rf ./my-cache-dir' EXIT


main/sorbet \
  --silence-dev-message \
  --print=index-tree \
  --cache-dir=my-cache-dir \
  --enable-experimental-rbs-signatures \
  test/cli/rbs-cache-dir/test.rb 2>&1

echo --------------------------------------------------------------------------

main/sorbet \
  --silence-dev-message \
  --print=index-tree \
  --cache-dir=my-cache-dir \
  test/cli/rbs-cache-dir/test.rb 2>&1
