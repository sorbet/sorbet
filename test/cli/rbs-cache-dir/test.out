--enable-experimental-rbs-signatures is incompatible with --cache-dir. Ignoring cache
No errors! Great job.
class <emptyTree><<C <root>>> < (::<todo sym>)
  class <emptyTree>::<C A><<C <todo sym>>> < (::<todo sym>)
    ::Sorbet::Private::Static.sig(::T::Sig::WithoutRuntime) do ||
      <self>.returns(<emptyTree>::<C Integer>)
    end

    def foo<<todo method>>(&<blk>)
      0
    end

    <runtime method definition of foo>
  end
end
--------------------------------------------------------------------------
No errors! Great job.
class <emptyTree><<C <root>>> < (::<todo sym>)
  class <emptyTree>::<C A><<C <todo sym>>> < (::<todo sym>)
    def foo<<todo method>>(&<blk>)
      0
    end

    <runtime method definition of foo>
  end
end
